src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -346,18 +346,25 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
 
         //#region automatic mining
         const nullResolved = Promise.resolve(null);
-        const mineAll = (maxTransactions: Capacity) =>
-          this.#isPaused() ? nullResolved : this.mine(maxTransactions);
+        const mineAll = (maxTransactions: Capacity, onlyOneBlock = false) =>
```

Comment:
update the mineAll function to actually utilize `blockchain.mine`'s `onlyOneBlock` flag